### ISSUE TITLE
Testing Mocks

### ISSUE THREAD

**Issue by justgook** [jodersky/flow#37]
Add some examples on how to create some testing mock for serial port..
For now I'm using (for local testing) `socat`:
```
# sudo socat -d -d pty,link=/dev/master,raw,echo=0,user=gook,group=staff  pty,link=/dev/slave,raw,echo=0,user=gook,group=staff
```
then I connect "terminal-sample" to the port `/dev/master` - and anything that is send to the `/dev/slave` got delivered to application.. 

would be nice get some examples on how to use that (or some other solution) for integration / unit testing, or/and how to crate mocks for Lib

**Comment by jodersky** [comment]
that's a great idea! Feel free to submit your suggestions as a PR, preferably into something like `Documentation/testing.md`. I'll see about adding some unit/integration tests, that's something the project desperately needs

**Comment by justgook** [comment]
added pull request #38

**Comment by jodersky** [comment]
closed by #38 , thanks
